design/workitems.go

```diff
@@ -41,6 +41,7 @@ var workItemRelationships = a.Type("WorkItemRelationships", func() {
 	a.Attribute("area", relationGeneric, "This defines the area this work item belongs to")
 	a.Attribute("children", relationGeneric, "This defines the children of this work item")
 	a.Attribute("space", relationSpaces, "This defines the owning space of this work item.")
+	a.Attribute("parent", relationGeneric, "This defines the parent of this work item.")
```

Comment:
It would be nice if you can instead add a new generic relation (not only for the parent), that stores the ID as a `uuid.UUID` instead of an `string` as it is done in the `relationGeneric` now. Then you can get rid of all the `uuid.FromString()` calls in your PR. You can then maybe also make the `ID` a required attribute and get rid of all the pointer `nil` checks because `ID` will then be a normal non-pointer member.

Reply:
yess ! done in https://github.com/fabric8-services/fabric8-wit/pull/1592/commits/92897142c26064ce0a23187ce40fd59c06c58bcb